Sources/Orbit/Components/SegmentedSwitch.swift

```diff
@@ -19,42 +21,54 @@ public struct SegmentedSwitch<Selection: Hashable, Content: View>: View {
     public var body: some View {
         FieldWrapper(label, message: message) {
             InputContent(message: message) {
-                content
-                    .environment(\.segmentSelection, selectionBinding)
-                    .overlay(separator)
-                    .backgroundPreferenceValue(IDPreferenceKey.self) { preference in
-                        selectionBackground(preference)
-                    }
-                    .fixedSize(horizontal: false, vertical: true)
+                HStack(spacing: 0) {
+                    content
+                        .allowsHitTesting(false)
+                        .frame(maxWidth: .infinity)
+                        .padding(.horizontal, .small)
+                        .padding(.vertical, verticalTextPadding)
+                }
+                .backgroundPreferenceValue(IDPreferenceKey.self) { preference in
+                    interactionBackground(preference)
+                }
+                .backgroundPreferenceValue(IDPreferenceKey.self) { preference in
+                    selectionBackground(preference)
+                }
+                .fixedSize(horizontal: false, vertical: true)
             }
         }
-        .accessibilityElement(children: .ignore)
+        .accessibilityElement(children: .contain)
         .accessibility(label: .init(label))
-        .accessibility(value: .init(selection.map(String.init(describing:)) ?? ""))
-        .accessibility(hint: .init(accessibilityHint))
-        .accessibility(addTraits: .isButton)
-        .accessibility(addTraits: selection != nil ? .isSelected : [])
+        .accessibility(hint: .init(message?.description ?? ""))
     }
 
-    @ViewBuilder var separator: some View {
-        (selection == nil ? borderColor : .clear)
-            .frame(width: borderWidth)
-            .frame(maxHeight: .infinity)
-            .padding(.vertical, BorderWidth.active)
+    @ViewBuilder func interactionBackground(_ preferences: [IDPreference]) -> some View {
+        GeometryReader { geometry in
+            ForEach(Array(preferences.enumerated()), id: \.element.id) { index, preference in
+                Color.clear
+                    .contentShape(Rectangle())
+                    .segmentFrame(bounds: preference.bounds, in: geometry)
+                    .onTapGesture {
```

Comment:
I thought we would go with using native Button, as it gives us "touchDown" highlight support and proper native accessibility. (The `isSelected` trait would still be kept, but the rest would be for free)

Reply:
But that highlight only applies to the button's title, not whatever is under it... at least from experimentation. I suppose we could create a custom button style, but the default one doesn't seem to work the way you think.

Reply:
So it's kind of possible to do this with an overlay in a custom button style.

Do we want the overlay to also "dim" the content...
https://github.com/kiwicom/orbit-swiftui/assets/12349477/5391ceef-3f96-4a45-8114-4e8bd3f9bd81

... or just the background
https://github.com/kiwicom/orbit-swiftui/assets/12349477/739185c4-d2ae-4502-bf21-faaddc6a7692
?

(Showing this here on `Tabs`, but the principle is the same)

Reply:
Yep, what I meant - a custom `ButtonStyle` that mimics the active/inactive background of what is under the title

True, the title itself would normally stay as-is with this approach, but in this case, that is probably what we want - to only decorate the background of the buttonm not the title itself. If we wanted to do that though, it would be much more difficult (we would again need to pass some `isSelected/isActive`  environment to the active component)

Reply:
Now we are saying: take the ViewBuilder content, obtain how many items are actually there and generate as many buttons based on their sizes (+ padding to each one). If we are fine with button background changes only, this should be all we need.